inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/util/QueryUtil.java

```diff
@@ -67,24 +66,106 @@
+        String exactMatchingTypedString = getExactMatchingQueryPart("exactTyped");
+
+        // Usually users are lazy and don't capitalize words,
+        // but then the exact matching wouldn't find anything.
+        String exactMatchingTypedStringCapitalized = getExactMatchingQueryPart("exactTypedCapitalized");
+
+        // Do the same for the mention
+        String exactMatchingMention = getExactMatchingQueryPart("exactMention");
+
+        String exactMatchingMentionCapitalized = getExactMatchingQueryPart("exactMentionCapitalized");
+
         String query = String.join("\n",
             "DEFINE input:inference 'instances'",
             SPARQL_PREFIX,
             "SELECT DISTINCT ?e2 ?altLabel ?label ?description WHERE",
             "{",
             "  {",
+            exactMatchingTypedString,
+            "  } ",
+            "  UNION",
+            "  {",
+            exactMatchingTypedStringCapitalized,
+            "  }",
+            "  UNION",
+            "  {",
+            exactMatchingMention,
+            "  }",
+            "  UNION",
+            "  {",
+            exactMatchingMentionCapitalized,
+            "  }",
+            "  FILTER EXISTS { ?e2 ?p ?v }",
+            "  FILTER NOT EXISTS ",
+            "  {",
+            "    VALUES ?topic {" + String.join(" ", WIKIMEDIA_INTERNAL,
+                WIKIMEDIA_PROJECT_PAGE, WIKIMEDIA_CATEGORY, WIKIMEDIA_DISAMBIGUATION_PAGE,
+                WIKIMEDIA_LIST_ARTICLE, WIKIMEDIA_TEMPLATE, WIKIMEDIA_NEWS_ARTICLE,
+                WIKIMEDIA_NAVIGATIONAL_TEMPLATE) +
+                "}",
+            "    ?e2 rdf:type ?topic",
+            "  }",
+            "  ?e2 rdfs:label ?label.",
+            "  FILTER ( lang(?label) = \"en\" )",
+            "}");
+
+        ValueFactory vf = SimpleValueFactory.getInstance();
+
+        TupleQuery tupleQuery = conn.prepareTupleQuery(QueryLanguage.SPARQL, query);
+        tupleQuery.setBinding("exactTyped", vf.createLiteral(aTypedString));
+        tupleQuery.setBinding("exactTypedCapitalized",
+            vf.createLiteral(StringUtils.capitalize(aTypedString)));
+        tupleQuery.setBinding("exactMention", vf.createLiteral(aMention));
+        tupleQuery.setBinding("exactMentionCapitalized",
+            vf.createLiteral(StringUtils.capitalize(aMention)));
+
+        tupleQuery.setBinding("descriptionIri", aDescriptionIri);
+        return tupleQuery;
+    }
+
+    /**
+     *
+     * This query retrieves candidates via exact matching of their labels and full-text-search
+     *
+     * @param aTypedString the unprocessed raw string the user searched for or
+     * @param aMention the words spanned by the mention, pre-processed
+     * @param aLimit maximum number of results
+     * @param aDescriptionIri KB-specific IRI that indicates a description
+     * @return a query to retrieve candidate entities
+     */
+    public static TupleQuery generateCandidateFullTextQuery(RepositoryConnection conn,
```

Comment:
Why is it public?

Reply:
I call it from a class in a different package.